server/src/main/java/org/elasticsearch/action/support/DestructiveOperations.java

```diff
@@ -37,6 +42,14 @@
     private volatile boolean destructiveRequiresName;
 
     public DestructiveOperations(Settings settings, ClusterSettings clusterSettings) {
+        if (REQUIRES_NAME_SETTING.exists(settings) == false) {
```

Comment:
This seems like it would be quit verbose, in that most users will end up getting this deprecation warning, and for most it will be noise. Could we be more selective in only emitting the deprecation when a wildcard destructive operation is done and the setting is using the default value?

Reply:
That's a very good idea. Thanks!